This patch contains vertical split patch and long term (up to 64 chars) patch.
---
diff -aur screen-4.2.1/ansi.c screen-4.2.1-arch/ansi.c
--- screen-4.2.1/ansi.c	2014-04-26 15:32:54.000000000 +0400
+++ screen-4.2.1-arch/ansi.c	2014-05-31 12:47:16.984557299 +0400
@@ -568,7 +568,7 @@
 		{
 		case '0': case '1': case '2': case '3': case '4':
 		case '5': case '6': case '7': case '8': case '9':
-		  if (curr->w_NumArgs < MAXARGS)
+		  if (curr->w_NumArgs >= 0 && curr->w_NumArgs < MAXARGS)
 		    {
 		      if (curr->w_args[curr->w_NumArgs] < 100000000)
 			curr->w_args[curr->w_NumArgs] =
diff -aur screen-4.2.1/os.h screen-4.2.1-arch/os.h
--- screen-4.2.1/os.h	2014-04-26 14:58:35.000000000 +0400
+++ screen-4.2.1-arch/os.h	2014-05-31 13:13:39.881220955 +0400
@@ -524,6 +524,6 @@
 /* Changing those you won't be able to attach to your old sessions
  * when changing those values in official tree don't forget to bump
  * MSG_VERSION */
-#define MAXTERMLEN	32
+#define MAXTERMLEN	64
 #define MAXLOGINLEN	256
 
diff -aur screen-4.2.1/pty.c screen-4.2.1-arch/pty.c
--- screen-4.2.1/pty.c	2014-04-26 14:58:35.000000000 +0400
+++ screen-4.2.1-arch/pty.c	2014-05-31 12:48:54.367889471 +0400
@@ -39,9 +39,11 @@
 #endif
 
 /* for solaris 2.1, Unixware (SVR4.2) and possibly others */
+#if defined(HAVE_SVR4_PTYS) && !defined(__APPLE__)
 #ifdef HAVE_STROPTS_H
 # include <sys/stropts.h>
 #endif
+#endif
 
 #if defined(sun) && defined(LOCKPTY) && !defined(TIOCEXCL)
 # include <sys/ttold.h>
diff -aur screen-4.2.1/resize.c screen-4.2.1-arch/resize.c
--- screen-4.2.1/resize.c	2014-04-26 14:58:35.000000000 +0400
+++ screen-4.2.1-arch/resize.c	2014-05-31 12:52:58.444555967 +0400
@@ -663,6 +663,18 @@
   if (wi <= 0 || he <= 0)
     wi = he = hi = 0;
 
+  if (wi > 1000)
+    {
+      Msg(0, "Window width too large, truncated");
+      wi = 1000;
+    }
+
+  if (he > 1000)
+    {
+      Msg(0, "Window height too large, truncated");
+      he = 1000;
+    }
+
   if (p->w_type == W_TYPE_GROUP)
     return 0;
 
